server/src/services/validation/validate.ts

```diff
@@ -132,11 +135,18 @@ function hardhatThrownFail(
 
     const displayText = hardhatError.errorDescriptor.title;
 
+    const errorFile = isHardhatSourceImportError(hardhatError)
+      ? decodeUriAndRemoveFilePrefix(
```

Comment:
Ugh, I tested on windows and we need to decode Uri and add a prefix of `/` on windows (i.e. `/c:/projects/example.sol` rather than `c:/project/example.sol`).

So maybe something like:
```typescript
function resolveErrorFilePath(projectBasePath: string, hardhatError: any): string {
  const errorPath = decodeUriAndRemoveFilePrefix(
    path.join(projectBasePath, hardhatError.messageArguments.from)
  )

  const osPath = os.platform() === "win32" ? `/${errorPath}` : errorPath

  return osPath
}
```